metrics/github/backfill.py

```diff
@@ -76,49 +77,34 @@ def pr_queue(prs, org, start, days_threshold=None):
                 >= timedelta(days=days_threshold)
             ]
 
-        suffix = f"_older_than_{days_threshold}_days" if days_threshold else ""
-        key = f"queue{suffix}"
+            name = f"queue_older_than_{days_threshold}_days"
 
-        log.info("%s | %s | %s | Processing %s PRs", key, day, org, len(prs_on_day))
-        with TimescaleDBWriter(GitHubPullRequests) as writer:
-            process_prs(writer, prs_on_day, day, f"queue{suffix}")
+            log.info(
+                "%s | %s | %s | Processing %s PRs", name, day, org, len(prs_on_day)
+            )
+            process_prs(writer, prs_on_day, day, name=name)
 
 
 def pr_throughput(prs, org, start):
     # get the previous DoW that we care about
     # loop through dates invoking the CLI entrypoint for each date
-    def next_weekday(d, weekday):
-        # 0 = Monday, 1=Tuesday, 2=Wednesday...
-        days_ahead = weekday - d.weekday()
-
-        if days_ahead <= 0:  # Target day already happened this week
-            days_ahead += 7
 
-        return d + timedelta(days=days_ahead)
+    first_day = date_from_iso(min([pr["created"] for pr in prs]))
+    days = list(iter_days(first_day, date.today()))
 
-    first_sunday = next_weekday(start - timedelta(days=7), 7)
-    next_sunday = next_weekday(date.today(), 7)
-    dates = [
-        first_sunday,
-        *list(iter_days(start, date.today(), step=timedelta(days=7))),
-        next_sunday,
-    ]
+    with TimescaleDBWriter(GitHubPullRequests) as writer:
+        for day in days:
+            opened_prs = [pr for pr in prs if date_from_iso(pr["created"]) == day]
+            log.info("%s | %s | Processing %s opened PRs", day, org, len(opened_prs))
+            process_prs(writer, opened_prs, day, name="prs_opened")
 
-    for day in dates:
-        start = day - timedelta(days=7)
-        end = day
-
-        prs_in_range = [
-            pr
-            for pr in prs
-            if date_from_iso(pr["created"]) >= start
-            and date_from_iso(pr["created"]) <= end
-        ]
-
-        key = "throughput"
-        log.info("%s | %s | %s | Processing %s PRs", key, day, org, len(prs_in_range))
-        with TimescaleDBWriter(GitHubPullRequests) as writer:
-            process_prs(writer, prs_in_range, day, name="throughput")
+            # FIXME: the selection below is capturing still-open PRs as closed
+            # on the final day, why??
```

Comment:
In answer to your `FIXME`. `date_from_iso` returns today's date if `pr['closed'] is None, which would explain why all open PRs are shown as being closed today.

Reply:
Hah, and I've already fixed it but forgot to remove the comment 🤦‍♂️